### DISCUSSION TITLE
Changing the returned user object from Twitter authentication

### DISCUSSION THREAD

**Discussion by sebastiancrossa** [nextauthjs/next-auth#1763] [Help] [answered]
**Your question**
I'm having some issues changing the scope of what is returned through the user object. The user object only returns name + profile picture, but I also need the users username returned.

**What are you trying to do**
I would like to have the username as a field too, but I haven't been able to pull it by adding scopes to the provider.

Example:
```js
Providers.Twitter({
      clientId: process.env.TWITTER_CLIENT_ID,
      clientSecret: process.env.TWITTER_CLIENT_SECRET,
      scope: "name username profile_image_url",
    }),
```

I've noticed that the Accounts database has an `accountProviderId` field, which can also be useful to pull the twitter username, but I'm not sure how I can do that either.

**Reproduction**
See above.

**Feedback**
- [ ] Found the documentation helpful
- [ ] Found documentation but was incomplete
- [ ] Could not find relevant documentation
- [ ] Found the example project helpful
- [x] Did not find the example project helpful

**Comment by balazsorban44** [comment, marked as answer]
have a look at the jwt and session callbacks

https://next-auth.js.org/configuration/callbacks

**Reply by sebastiancrossa** [reply]
Have been playing around with the callbacks for a while, but is there a way to access the accounts table?

**Reply by sebastiancrossa** [reply]
Quick update - I finally found all of the information needed, and it was through the signIn callback, specifically the profile parameters :)

**Reply by bguan2020** [reply]
can I ask how you did this? Have the same issue, I want useSession to return username in addition to email and name. @sebastiancrossa
